sources/packages/api/src/services/institution-location/institution-location.service.ts

```diff
@@ -94,25 +146,29 @@ export class InstitutionLocationService extends RecordDataModelService<Instituti
   }
 
   async getInstitutionLocation(
-    institutionId: number,
     locationId: number,
+    institutionId?: number,
   ): Promise<InstitutionLocation> {
-    return this.repo
+    const query = this.repo
       .createQueryBuilder("institution_location")
       .select([
         "institution_location.name",
         "institution_location.data",
-        "institution.institutionPrimaryContact",
         "institution_location.id",
         "institution_location.institutionCode",
         "institution_location.primaryContact",
       ])
-      .leftJoin("institution_location.institution", "institution")
-      .where("institution.id = :id and institution_Location.id = :locationId", {
-        id: institutionId,
+      .where("institution_Location.id = :locationId", {
         locationId: locationId,
-      })
-      .getOne();
+      });
+    if (institutionId) {
+      query
+        .leftJoin("institution_location.institution", "institution")
```

Comment:
This must be a `innerJoin`, is was wrong in the existing code.

Reply:
Cool, I thought of it when I created the query, I forgot to discuss in our call though 😄